alacritty/src/display/window.rs

```diff
@@ -287,47 +262,104 @@ impl Window {
 
     #[cfg(not(any(target_os = "macos", windows)))]
     pub fn get_platform_window(
+        event_loop: &dyn ActiveEventLoop,
         identity: &Identity,
         window_config: &WindowConfig,
+        activation_token: Option<String>,
         #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))] x11_visual: Option<
             X11VisualInfo,
         >,
     ) -> WindowAttributes {
-        #[cfg(feature = "x11")]
-        let icon = {
-            let mut decoder = Decoder::new(Cursor::new(WINDOW_ICON));
-            decoder.set_transformations(png::Transformations::normalize_to_color8());
-            let mut reader = decoder.read_info().expect("invalid embedded icon");
-            let mut buf = vec![0; reader.output_buffer_size()];
-            let _ = reader.next_frame(&mut buf);
-            Icon::from_rgba(buf, reader.info().width, reader.info().height)
-                .expect("invalid embedded icon format")
+        let activation_token = if let Some(activation_token) = activation_token
+            .map(ActivationToken::from_raw)
+            .or_else(|| event_loop.read_token_from_env())
+        {
+            log::debug!("Activating window with token: {activation_token:?}");
+            // Remove the token from the env.
+            startup_notify::reset_activation_token_env();
+            Some(activation_token)
+        } else {
+            None
         };
 
-        let builder = WinitWindow::default_attributes()
-            .with_name(&identity.class.general, &identity.class.instance)
-            .with_decorations(window_config.decorations != Decorations::None);
+        match event_loop.display_handle().unwrap().as_raw() {
+            #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+            RawDisplayHandle::Xlib(_) | RawDisplayHandle::Xcb(_) => {
+                Self::get_x11_window(identity, window_config, activation_token, x11_visual)
+            },
+            #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+            RawDisplayHandle::Wayland(_) => {
+                Self::get_wayland_window(identity, window_config, activation_token)
```

Comment:
Is it really necessary to create a window for each individual platform now? That seems like a horrible API. Wasn't the whole purpose of winit to abstract over this?

Reply:
Sounds like yet another breaking change coming up? :smiling_face_with_tear: